libsolidity/ast/ASTPrinter.cpp

```diff
@@ -579,8 +582,11 @@ void ASTPrinter::printSourcePart(ASTNode const& _node)
 	if (!m_source.empty())
 	{
 		SourceLocation const& location(_node.location());
-		*m_ostream << indentation() << "   Source: "
-				   << escaped(m_source.substr(location.start, location.end - location.start), false) << endl;
+		*m_ostream <<
```

Comment:
I think for reviewability we should split this PR up. One simple change is replacing `escaped` with `jsonCompactPrint` and removing `escaped`.

Reply:
Having a second look, are you sure the substring from the source is JSON?

Reply:
No, it is not json, it is a pure string, but the json formatter will escape it in exactly the same way it was escaped before.

Reply:
That's weird, because string on the root level is invalid according to the JSON spec.

Reply:
This is just a restriction of the parser. Json::Value is an arbitrary json value, including any internal value.